qiskit/__init__.py

```diff
@@ -94,14 +96,16 @@ def __init__(self):
     def __bool__(self):
         if self.aer is None:
             try:
-                from qiskit.providers import aer
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    from qiskit.providers import aer
 
                 self.aer = aer.Aer
                 warnings.warn(
-                    "The qiskit.Aer entry point will be deprecated in a future release and "
-                    "subsequently removed. Instead you should use this "
-                    "directly from the root of the qiskit-aer package.",
-                    PendingDeprecationWarning,
+                    "The qiskit.Aer entry point is deprecated and will be removed in a future "
+                    "release. Instead you should use this directly from the root of the "
+                    "qiskit-aer package: 'qiskit_aer.Aer'",
+                    DeprecationWarning,
```

Comment:
Removing the `qiskit.Aer` entry point isn't _technically_ needed to remove the namespace packaging.  Are we certainly resolved to remove this too, and not just change the import to `from qiskit_aer import Aer`?

I'm ok either way. If we're going to deprecate it, then all at once with `qiskit.providers.aer` is the best way, I think - it's not ideal for users to deprecate, but better than doing it in a half measure.

Reply:
I figured we'd want to do it all at once, basically try to separate the packages as much as possible so aer is a strictly a downstream consumer of terra. In practice there are still the fake backends and some algorithms/opflow code where there is an optional dependency

Reply:
I don't feel _great_ about removing the `Aer` helper, but that said, I'm completely fine with us removing `IBMQ` when it comes to it, so maybe it's just a bit of weirdness in my head.  If we (and higher up) are resolved to completely separate the packages, it does make sense to do it all at once rather than stringing it out.